presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddLocalExchanges.java

```diff
@@ -820,6 +823,38 @@ public PlanWithProperties visitIndexJoin(IndexJoinNode node, StreamPreferredProp
             return rebaseAndDeriveProperties(node, ImmutableList.of(probe, index));
         }
 
+        @Override
+        public PlanWithProperties visitMergeJoin(MergeJoinNode node, StreamPreferredProperties parentPreferences)
```

Comment:
will adding a local merging exchange before the merge join be sufficient to support non-grouped execution?